### PR TITLE
Modify schedule class to create schedule objects

### PR DESCRIPTION
**Pull request by anavilohia**
This PR

- Updates schedule class - more details below
- Updates schedule class unit tests
- Comments out Schedule unit tests - will add new ones in the next PR
- Passes PMD check
- Passes checkstyle
- Modifies Jacoco coverage to 68% 

Schedule class is now constructed only with allTasks as they already contain ResourceTypes which contain resources.
The createSchedule method returns a list of doable tasks and the resources assigned to the task until task end time.

### CODE REVIEW

**Review by janelim0414 — DISMISSED** [review]
Looks great! Thanks for working on this. Will take a look at the next PR for the unit tests.

File: LiveSched/src/test/java/dev/coms4156/project/livesched/ScheduleUnitTests.java [hunk 1]
```diff
@@ -23,145 +25,134 @@
...
+        () -> new Schedule("  ", mockTasks, maxDistance),
         "Schedule constructor should throw an exception if scheduleId is empty.");
     assertEquals("Schedule ID cannot be null or empty.", exception.getMessage());
 
-    exception = assertThrows(IllegalArgumentException.class, 
-        () -> new Schedule(scheduleId, null, maxDistance),
-        "Schedule constructor should throw an exception if resourceTypes is null.");
-    assertEquals("Resource types list cannot be null.", exception.getMessage());
+    exception = assertThrows(IllegalArgumentException.class,
+        () -> new Schedule(testScheduleId, null, maxDistance),
```
Comment (janelim0414) [inline LiveSched/src/test/java/dev/coms4156/project/livesched/ScheduleUnitTests.java:81]: Should we also consider a case where task list is empty? This also makes me think whether we should allow real-time processing of incoming task data. I think this could be done by using a separate thread that receives incoming tasks from the network and adds to the queue -- what do you think?

File: LiveSched/src/test/java/dev/coms4156/project/livesched/ScheduleUnitTests.java [hunk 2]
```diff
@@ -23,145 +25,134 @@
...
-    resourceTypes.add(testResourceType);
-    
-    testSchedule = new Schedule(scheduleId, resourceTypes, maxDistance);
-    
-    Map<ResourceType, Integer> resourceList = new HashMap<>();
-    resourceList.put(testResourceType, 1);
-    LocalDateTime startTime = LocalDateTime.now().plusHours(1);
-    LocalDateTime endTime = startTime.plusHours(2);
-    testTask = new Task("TestTask", resourceList, 3, startTime, endTime, 40.7128, -74.0060);
+    mockTask1 = mock(Task.class);
```
Comment (janelim0414) [inline LiveSched/src/test/java/dev/coms4156/project/livesched/ScheduleUnitTests.java:46]: Love how you mocked the behaviors of the class methods for simplicity :)

**Review by janelim0414 — APPROVED** [review]
Should we also consider doing an integration test to make sure that schedules are created with the right priorities?

File: LiveSched/src/test/java/dev/coms4156/project/livesched/ScheduleUnitTests.java (hunk 1 above)
Comment (janelim0414) [reply LiveSched/src/test/java/dev/coms4156/project/livesched/ScheduleUnitTests.java:81]: In that case, we won't need to throw an error when task list is empty. `createSchedule` also won't need to iterate through the task list but instead process task one by one whenever there is an element in the queue. We'd have to keep a separate thread for `createSchedule` as well.

File: LiveSched/src/test/java/dev/coms4156/project/livesched/ScheduleUnitTests.java (hunk 2 above)
Comment (anavilohia) [reply LiveSched/src/test/java/dev/coms4156/project/livesched/ScheduleUnitTests.java:46]: Yes, will do integration tests to check

File: LiveSched/src/test/java/dev/coms4156/project/livesched/ScheduleUnitTests.java (hunk 1 above)
Comment (anavilohia) [reply LiveSched/src/test/java/dev/coms4156/project/livesched/ScheduleUnitTests.java:81]: Real time processing of incoming tasks sounds great! we can add it to the Jira tasks to address if/when we have time. I think createSchedule may still need to iterate if we want to create a schedule of multiple tasks mapped to resources, addressing only one at a time seems to beat the purpose of promoting efficiency through our service. But if I'm missing something there happy to understand better over discord!

Comment (anavilohia) [reply LiveSched/src/test/java/dev/coms4156/project/livesched/ScheduleUnitTests.java:81]: Also, will add a test for empty task list in next PR
